NBitcoin/Network.cs

```diff
@@ -2562,7 +2566,8 @@ public IBitcoinString Parse(string str, Type? targetType)
 						candidate = new BitcoinWitPubKeyAddress(str.ToLowerInvariant(), bytes, this);
 					if (witVersion == 0 && bytes.Length == 32 && type == Bech32Type.WITNESS_SCRIPT_ADDRESS)
 						candidate = new BitcoinWitScriptAddress(str.ToLowerInvariant(), bytes, this);
-
+					if (witVersion == 1 && bytes.Length == 32)

```

Comment:
Aren't we missing this?

```suggestion
					if (witVersion == 1 && bytes.Length == 32 && type == Bech32Type.TAPROOT_ADDRESS)
```

Reply:
yes, I need a bunch of tests as well. I want to do this week.